### DISCUSSION TITLE
GTK: Numeric icons to indicate tab position

### DISCUSSION THREAD

**Discussion by jcollie** [ghostty-org/ghostty#2624] [Feature Requests, Ideas] [answered]
As discussed on Discord: https://discord.com/channels/1005603569187160125/1304093836439457842

The proposal is to add icons to the 1st 10 tab headers to indicate the tab position, making it easier to switch tabs using keybindings (`alt+one`..`alt+zero`). This is what it would look like:

![Screenshot_From_2024-11-07_23-39-18](https://github.com/user-attachments/assets/8bfc0602-8eb8-4cf8-9f6f-d1b502147d59)

Initially I was using icons taken from https://pictogrammers.com/ but I liked ones that I drew better, but I'm sure that a more experienced designer could improve on my efforts.

This would be gated with a configuration entry, which would default to not showing any icons.

**Comment by jcollie** [comment]
The implementation will be made easier by the refactoring brought in by #2529 .

**Comment by gpanders** [comment]
>This would be gated with a configuration entry, which would default to not showing any icons.

Does this need to be configurable? And if so, why disable it by default?

We have this feature on macOS (it was my very first contribution to Ghostty!) and it's not configurable. Personally, I don't think it needs to be (in the interest of avoiding configuration overload).

**Reply by jcollie** [reply]
"need to be"? no, but on the discord discussion it was suggested to make it so. I don't have strong feelings either way.

**Reply by mitchellh** [reply]
Yeah I originally suggested it because it's not a native widget. Depending how it looks/feels though perhaps you're right.

**Reply by gpanders** [reply]
>Yeah I originally suggested it because it's not a native widget. Depending how it looks/feels though perhaps you're right.

I think this is a good distinction. The labels used in macOS are fairly subtle and feel native.

Does GTK require using icons? Can we not just use simple text?

I dearly miss the tab labels when I use Linux so I'm excited to see this feature being discussed (I looked into doing this in GTK briefly, but couldn't find any straightforward way to add a label/secondary text to the tab).

**Reply by jcollie** [reply]
> Does GTK require using icons? Can we not just use simple text?

I wouldn't say that icons are _required_ but prepending the tab title text with unicode like ① or ⑨ seemed a little hackier to me. Plus using an icon _is_ using a native feature of the libadwaita tab pages (it's basically 2 function calls - one call to load an icon and one call to set the icon on the page). There are other things that could be done with icons, like setting them to a colored icon to indicate membership in a group or something like that.

**Reply by gpanders** [reply]
>I wouldn't say that icons are required but prepending the tab title text with unicode like ① or ⑨ seemed a little hackier to me.

Is ① a requirement instead of plain ole `1`? The circle feels like an opinionated aesthetic choice. That's not necessarily bad, but I don't think it's part of the native design language of the platform (I don't recall seeing it anywhere in GNOME or other GTK apps at least).

Opinionated aesthetic choices are more likely to require configuration to disable them. So if we go that route then perhaps we do want a config option for this.

**Reply by jcollie** [reply]
I think that the circle distinguishes it from being just part of the title text.

**Comment by mitchellh** [comment, marked as answer]
Opened issue #2629 to accept this.
